smarts/core/controllers/lane_following_controller.py

```diff
@@ -422,8 +425,10 @@ def _update_target_lane_if_reached_end_of_lane(
         # When we reach the end of our target lane, we need to update it
         # to the next best lane along the path
         state = controller_state
-        paths = sensor_state.mission_planner.waypoint_paths_on_lane_at(
-            vehicle.pose, state.target_lane_id, lookahead=2
+        plan = sensor_state.plan
+        lane = plan.road_map.lane_by_id(state.target_lane_id)
+        paths = lane.waypoint_paths_at_point(
+            vehicle.pose, lookahead=2, route=plan.route
         )
```

Comment:
actually, I had planned to circle back to change it to take a `Point` instead of a `Pose` (we don't use the heading anywhere in it), but I forgot.  In any case, I agree the name should match the argument.

Reply:
So I just changed the method name to `waypoint_paths_for_pose`.  I ended up liking that better than what I wrote in the above comment because it allows for some implementations to use the heading to influence what waypoints to return, which might be useful.